docs/motivation.md

```diff
@@ -0,0 +1,550 @@
+# The Verona language: Core type system
+
+Verona is a language with a behaviour-oriented concurrency model and a region abstraction for ownership.
+These are fundamentally new abstractions in language design and are covered separately.
+This document aims to cover the parts of the type system that are similar to (though in some places diverge in interesting ways from) other languages.
+
+Verona is an object oriented reference capability language with structural
+subtyping and bounded polymorphism. This document aims to describe the aspects
+of the type system that are independent from both the concurrency model and
+reference capabilities.
+
+## Structural type system
+
+A type system is all about managing a complexity budget. Make it too complicated
+and programmers will find it difficult to use and be less productive. Make it too
+simple and it will give you less guarantees to lean on as a programmer.
+
+We opt to use a structural type system since we beleive it simplifies some parts
+of reasoning, e.g. we don't need co/contra-variance notation on type parameters.
+This allows us to spend this complexity elsewhere, e.g. introducing reference
+capabilities to enforce isolation and region topology.
+
+We do however treat classes nominally and have opted to not have inheritance
+subtyping. This means that there is never any subtyping between two distinct
+classes.
+
+### Traits
+Verona is a structural type system based on traits, i.e. types can be described
+using combination of methods on an object. E.g. we can describe a type
+implementing a `toString` method as
+```verona
+type ToString = {
+    toString(self: Self): String
+}
+```
+A type describing numbers implementing addition and subtraction can be described with
+```verona
+type Number = {
+    add(self: Self, other: Number): Number
+    sub(self: Self, other: Number): Number
```

Comment:
This is an opportunity to use symbolic function names:

type Number =
{
  +(self: Self, other: Number): Self
  -(self: Self, other: Number): Self
}

Also, a "self types" section is needed.